openest/generate/v1/fake.py

```diff
@@ -29,11 +29,13 @@ def generate_bilinear(fips, yyyyddd, values):
             relative[relative < 0] = relative[relative < 0] * slope_neg
             result = baseline + np.sum(relative)
             if not np.isnan(result):
-                yield (year, result)
+                yield year, result
 
     return generate_bilinear
 
-def make_print_bymonthdaybins(model, func=lambda x: x, weather_change=lambda temps: temps - 273.15, limits=[-40, 100], pval=.5):
+def make_print_bymonthdaybins(model, func=lambda x: x, weather_change=lambda temps: temps - 273.15, limits=None, pval=.5):
+    if limits is None:
+        limits = [-40, 100]
```

Comment:
This just seems messy. Personally, I think that lists and dicts in arguments are stylistically preferable as long as you aren't going to mutate them, but I can appreciate that some people don't want them. But in this case, just change limits to a tuple.

Reply:
This isn't a style issue. This is a definite anti-pattern in Python. Removing mutables from args completely eliminates the chance that we have another Algeria bug-style mistake from shared mutables in our args. This would be embarrassing for all and it's easily avoidable.

 Rant aside. I think swapping this with a tuple is a fine suggestion. I'll make this change!

Reply:
Resolved with c48b9d0. I switched default arg back and converted the list to a tuple.